1-Draft/RFCnnnn-Alias-Policy.md

```diff
@@ -0,0 +1,108 @@
+
+The purpose of this document is to establish a new policy for the creation and use of aliases to ensure:
+
+* Reliable, predictable behavior of scripts and modules
+* Increase readability and maintainability of scripts and modules
+* UX enhancements in interactive sessions
+
+# Problem description
+
+1. Windows PowerShell has many aliases that conflict with the names of standard Unix utilities.
+
+    These aliases was removed from PowerShell Core.
+    This causes a backward compatibility issue when Windows users do not find the familiar aliases in the interactive session.
+    It also disrupts the operation of custom scripts that use these aliases.
+
+1. Now there are dozens of modules developed by MSFT and thousands of modules of third developers. This causes a name conflict.
+
+    Developers try to create unique cmdlet names.
+    They can use module prefixes to resolve name conflicts.
+
+    On the other hand, developers of the best intentions can complement their modules cmdlet aliases.
+    At the same time, users can create their own aliases.
+
+    With a large number of existing modules and cmdlets and a wide variety of custom aliases, these short names inevitably conflict with each other.
+
+    Moreover, the behavior becomes unpredictable because each subsequent download module or script can change any alias.
+    Either the user can destroy a script behavior by creating a new alias or by deleting an existing one.
+
+    It is even more unpredictable and dangerous to create scripts in the assumption that there are some predefined (standard) aliases because they can be changed by the user or script at any time.
+
+    Modern versions of ScriptAnalyzer already recommend replacing aliases in scripts with their full names.
+
+    >The definition of aliases in modules or scripts and their automatic import becomes an undesirable and dangerous practice.
+
+1. UX in interactive sessions.
+
+    Having many cmdlets and parameters with long names can make it difficult to enter them in interactive sessions.
+    Users who typically use standard settings are typically satisfied with the ```IntellySense``` capabilities.
+    On the other hand, some users actively using interactive sessions may still be annoyed by the need to type names.
+    These users should be able to upload their familiar and useful aliases sets to their interactive sessions.
+    Given the limited functionality of aliases, these users will also configure their environment by executing commands by importing their own functions and various modules using PowerShell profiles.
+    It should be noted that we are planning to add a domain-specific language (DSL). This will provide users with much greater customization capabilities than aliases.
+
+# PowerShell alias policy
+
+1. PowerShell implements standard aliases in all sessions including interactive:
+
+    * % -> ForEach-Object
+    * foreach -> ForEach-Object
+    * ? -> Where-Object
+    * where -> Where-Object
```

Comment:
Should we add `* select -> Select-Object` ?

Reply:
I believe that these are just examples -- but perhaps we should have a discussion about them. If we're going to create a list of "standard" aliases that will always be there, that list becomes almost sacrosanct, and none of these "standard" aliases should ever be removed. Right?

In any case, this particular list is problematic. I agree that the aliases for the `*-Object` commands are some of the most used, but you would need to include not just `select` but also `sort`, `tee`, and `group`, as well as `compare` and `measure` ...

However, `where` and `sort` are probably **already** on the chopping block (as-in, should have been removed already, based on my understanding of the new rules) because they clobber _built-in_ command-line executables in Windows. If those two need to be removed to an optional set, then **I don't see any reason to include _any_ of these outside of an optional set.**

Given that, I think probably **there _are no standard aliases_**.